### PR TITLE
prov/efa: use same strategy to select RTM for neuron and cuda

### PR DESCRIPTION
**Pull request by wzamazon**
This patch renamed

    rxr_msg_select_rtm_for_cuda()

to

    rxr_msg_select_rtm_for_hmem()

and used it for neuron memory too.

Signed-off-by: Wei Zhang <wzam@amazon.com>

### CODE REVIEW

File: prov/efa/src/rxr/rxr_msg.c [hunk 1]
```diff
@@ -56,18 +56,15 @@
  */
 
 /**
- * @brief select a two-sided protocol for the send operation when send buffer is on cuda memory
+ * @brief select a two-sided protocol for the send operation when send buffer is on hmem memory
  *
  * @param[in]		rxr_ep		endpoint
  * @param[in]		tx_entry	contains information of the send operation
  * @return		the RTM packet type of the two-sided protocol.
- *                      Four types of protocol can be used: eager, medium, longread or runtread.
```
Comment (wckzhang) [inline prov/efa/src/rxr/rxr_msg.c:64]: Why did you remove these comments? (also line 67

**Review by wckzhang — APPROVED** [review]

File: prov/efa/src/rxr/rxr_msg.c [hunk 2]
```diff
@@ -154,10 +151,9 @@ int rxr_msg_select_rtm(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_entry, int
 		return RXR_EAGER_MSGRTM_PKT + tagged;
 	}
 
-	if (use_p2p && efa_mr_is_cuda(tx_entry->desc[0]))
-		return rxr_msg_select_rtm_for_cuda(rxr_ep, tx_entry);
+	if (use_p2p && efa_mr_is_hmem(tx_entry->desc[0]))
```
Comment (shijin-aws) [inline prov/efa/src/rxr/rxr_msg.c:154]: I do not think we will do runting read for habana?

Comment (shijin-aws) [reply prov/efa/src/rxr/rxr_msg.c:154]: I think I could just insert the habana block before this `if`

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_msg.c:154]: Hababa is not in the code base yet. so I did not include it.

After habana is added, I think it is better to handle it inside the function `select_rtm_for_hmem`.

Comment (shijin-aws) [reply prov/efa/src/rxr/rxr_msg.c:154]: Hmm I think I could just insert the habana block before this `if`.

Comment (shijin-aws) [reply prov/efa/src/rxr/rxr_msg.c:154]: @wzamazon That sounds good. Thank you!

**Review by shijin-aws — APPROVED** [review]
